ecl/hql/hqlattr.cpp

```diff
@@ -2550,9 +2564,10 @@ IHqlExpression * calcRowInformation(IHqlExpression * expr)
         {
             retrieveRowInformation(info, ds);
 
-            __int64 limit = getIntValue(expr->queryChild(1), 0);
+            IHqlExpression * limitExpr = expr->queryChild(1);
+            __int64 limit = getIntValue(limitExpr, 0);
             if ((limit != 0) && !isGrouped(expr))
-                info.applyChoosen(limit, isLocalActivity(expr));
+                info.applyChoosen(limitExpr, limit, isLocalActivity(expr));
             else
                 info.limitMin(limit);
```

Comment:
Would it be possible to common up this code? 

i.e. replace lines 2568-2572 with
info.applyChoosen(limitExpr, isLocalActivity(expr), !isGrouped(expr))

and 
move the "if ((limit != 0) && !isGrouped(expr))" to within applyChoosen()?